tests/console/validate_file_system.pm

```diff
@@ -22,16 +22,35 @@ use Test::Assert ':all';
 use partitions_validator_utils 'validate_partition_table';
 
 sub run {
-    my $test_data  = get_test_suite_data();
-    my %partitions = %{$test_data->{file_system}};
+    my $test_data            = get_test_suite_data();
+    my $partitions           = $test_data->{file_system};
+    my $encrypted_partitions = $test_data->{encrypted_filesystem};
+
+    select_console 'root-console';
 
     validate_partition_table({device => $test_data->{device}, table_type => $test_data->{table_type}});
 
-    foreach (keys %partitions) {
-        record_info("Check fs $_", "Verify the '$_' partition filesystem is '$partitions{$_}'");
-        my $fstype = script_output("df -PT $_ | grep -v \"Filesystem\" | awk '{print \$2}'");
-        assert_equals($partitions{$_}, $fstype,
-            "File system on '$_' partition does not correspond to the expected one");
+    if (defined $partitions) {
+        foreach (keys %{$partitions}) {
+            record_info("Check fs $_", "Verify the '$_' partition filesystem is '$partitions->{$_}'");
+            my $fstype = script_output("df -PT $_ | grep -v \"Filesystem\" | awk '{print \$2}'");
+            assert_equals($partitions->{$_}, $fstype,
+                "File system on '$_' partition does not correspond to the expected one");
+        }
+    }
+
+    if (defined $encrypted_partitions) {
+        foreach (@{$encrypted_partitions}) {
+            record_info("Check size", "Verify encrypted partition size");
+            assert_script_run("lsblk -n $test_data->{device} | grep '$_->{size}'",
```

Comment:
shouldnt it be $encrypted_partitions->[0]->{partition}  (/dev/vdb1). This will return any partition with the given size. Unlikely in this scenario but i cant see why not use it as it is already given in test_data.

Reply:
it works anyway, but I wanted to be more specific and forgot at the end, it is enough with `$_->{partition}`. I will fix it.

Reply:
i know that works anyway. it could ignored for now. i just stuck on it. thanks anyway